crates/prover/Cargo.toml

```diff
@@ -82,6 +82,8 @@ required-features = ["export-tests"]
 
```

Comment:
I don't like how the prover crate now has a "cuda" and "network feature that don't do anything at all. On second thought, it's even kind of weird that the ProverMode lives in the prover crate in the first place. I feel like the prover crate should pretty much solely be concerned with generating different kinds of proofs on-device, and not really think about anything else. 

cc @ctian1 @ratankaliani for second opinions

@leruaa why did you need to move the provermode here in the first place?